test/e2e/e2e.go

```diff
@@ -158,6 +158,68 @@ var _ = Describe("validation", func() {
 			})
 		})
 	})
+
+	Context("MetalLB contains incorrect data", func() {
+		Context("MetalLB has incorrect name", func() {
+
+			var metallb *metallbv1alpha.Metallb
+			var metallbCRExisted bool
+
+			BeforeEach(func() {
+				metallb = &metallbv1alpha.Metallb{}
+				err := loadMetallbFromFile(metallb, MetallbCRFile)
+
+				metallb.SetName("incorrectname")
+				Expect(err).ToNot(HaveOccurred())
+
+				metallbCRExisted = true
+				err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: metallb.Namespace, Name: metallb.Name}, metallb)
+				if errors.IsNotFound(err) {
+					metallbCRExisted = false
+					Expect(testclient.Client.Create(context.Background(), metallb)).Should(Succeed())
+				} else {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+
+			AfterEach(func() {
+				if !metallbCRExisted {
+					err := testclient.Client.Delete(context.Background(), metallb)
+					Expect(err).ToNot(HaveOccurred())
+					// Check the MetalLB custom resource is deleted to avoid status leak in between tests.
+					Eventually(func() bool {
+						err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: metallb.Namespace, Name: metallb.Name}, metallb)
+						if errors.IsNotFound(err) {
+							return true
+						}
+						return false
+					}, 5*time.Minute, 5*time.Second).Should(BeTrue(), "Failed to delete MetalLB custom resource")
+				}
+			})
+			It("should have not deploy metallb resources", func() {
```

Comment:
should not deploy or should not have deployed

Reply:
done